packages/datafile-manager/src/reactNativeAsyncStorageCache.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2020, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getLogger } from '@optimizely/js-sdk-logging'
+import AsyncStorage from '@react-native-community/async-storage'
+
+import PersistentKeyValueCache from './persistentKeyValueCache'
+
+const logger = getLogger('DatafileManager')
+
+export default class ReactNativeAsyncStorageCache implements PersistentKeyValueCache {
+
+  get(key: string): Promise<Object | null> {
+    return new Promise((resolve, reject) => {
+      AsyncStorage.getItem(key).then((val: string | null) => {
+        if (val) {
+          try {
+            resolve(JSON.parse(val))
+          } catch (ex) {
+            logger.error('Error Parsing Object from cache - %s', ex)
+            reject(ex)
+          }
+        } else {
+          resolve(null)
+        }
+      }).catch(reject)
+    })
```

Comment:
I still don't think constructing promise is necessary, better to chain off the `getItem` promise. I think this behaves the same:
```suggestion
   return AsyncStorage.getItem(key).then((val: string | null) => {
     if (val) {
       return JSON.parse(val);
     }
     return null;
   }).catch((ex) => {
     logger.error('Error Parsing Object from cache - %s', ex);
     throw ex;
   });
```
Usually it is best to chain off existing promises, instead of creating them.